### PR TITLE
added backward charge estimation to losses

### PR DESCRIPTION
**Pull request by AlexTeslenko**
added backward charge estimation to losses

### CODE REVIEW

**Review by vhavrylov-ncube — COMMENTED** [review]
Looks good, but it looks like we have no recursive computing of a backward pas from error layer until a particular node.

File: libs/ml/include/ml/ops/ops.hpp [hunk 1]
```diff
@@ -170,7 +170,7 @@ class Ops
    * @return estimated charge amount, necessary for performing a backward pass on data of given
    * shapes.
    */
-  virtual OperationsCount ChargeBackward() const
+  virtual OperationsCount ChargeBackward()
```
Comment (vhavrylov-ncube) [inline libs/ml/include/ml/ops/ops.hpp:173]: Are you sure about removing the const specifier? This method is intended to be a getter-only, e.g. a const one.

**Review by vhavrylov-ncube — CHANGES_REQUESTED** [review]
I would vote for keeping getters with "const" specifier

**Review by kublai-baykaner — APPROVED** [review]

**Review by vhavrylov-ncube — APPROVED** [review]
LGTM

**Pull request merged**